_posts/2019-06-12-problem-nextcloud-global-dependencies.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: single
+title: The Problem of Using Nextcloud's Globally Loaded Dependencies
+comments: true
+date: 2019-06-12
+last_modified_at: 2019-06-12
+tags:
+  - nextcloud
+  - development
+  - javascript
+  - foss
+---
+
+Many Nextcloud apps that serve a front-end component have been rewritten in Vue.js recently. With this change to a more modern framework and concept for building user interfaces, more apps have also started using webpack to bundle scripts, styles and even icons. In this post I would like to reflect on why this is important and why more apps should do it.
+
+## 🕵️ Writing Nextcloud app -- the old approach
+
+Before bundlers like webpack became popular in the Nextcloud community, most apps did not use any bundling for their assets. This means scripts and style sheets were included via HTML's `<script>` and `<style>` tags.
+This style of front-end programming comes with a few problems and limitations.
+
+### 🐌 Increased number of requests
+With the modular architecture of Nextcloud and having loaded many apps, this design results in numerous requests for a page load without a populated cache. The more files (modules) an app has, the more requests the browser has to wait for before the page can fully work.
+
+Some developers start with small modules, others split larger onces when they refactor they code that might otherwise become unmaintainable. However, if you are aware of the problem of too many requests on page load, you might rather inline a small module than separating it into a self-contained module. Eventually, this leads to a lower code quality and less maintainable code. [js.js as example]
```

Comment:
when they refactor their code? :)

Reply:
yes! and I should have replaced the `[js.js as example]` as well :see_no_evil: